src/libasr/pass/intrinsic_functions.h

```
@@ -2519,8 +2519,13 @@ namespace Ifix {
 namespace Idint {
 
     static ASR::expr_t *eval_Idint(Allocator &al, const Location &loc,
-            ASR::ttype_t* /*arg_type*/, Vec<ASR::expr_t*> &args, diag::Diagnostics& /*diag*/) {
+            ASR::ttype_t* /*arg_type*/, Vec<ASR::expr_t*> &args, diag::Diagnostics& diag) {
         int val = ASR::down_cast<ASR::RealConstant_t>(expr_value(args[0]))->m_r;
+        int kind = ASRUtils::extract_kind_from_ttype_t(ASR::down_cast<ASR::RealConstant_t>(args[0])->m_type);
```

```suggestion
        int kind = ASRUtils::extract_kind_from_ttype_t(expr_type(args[0]));
```

---

Also, I'm not sure if we should be throwing error here. It would have been better if we could throw this error earlier. In the verify_args and create_idint function.

---

Do we have support for real64 in python file wherein we use to automate verify_args and create functions at this point of time? I think we dont